src/history/HistoryArchive.cpp

```diff
@@ -289,6 +290,12 @@ HistoryArchiveState::HistoryArchiveState(uint32_t ledgerSeq,
         auto& level = buckets.getLevel(i);
         b.curr = binToHex(level.getCurr()->getHash());
         b.next = level.getNext();
+        if (clearFutureBuckets)
+        {
+            // Future buckets aren't needed in the archives
+            // post protocol 12.
```

Comment:
use `Bucket::FIRST_PROTOCOL_SHADOWS_REMOVED` not 12

Reply:
hmm, it's quite verbose that way. "Protocol 10" and "Protocol 11" are used consistently in the comments/tests throughout the codebase, I don't see how this is any different.